src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -155,7 +156,7 @@ public BitArray(int[] values)
             }
 
             m_array = new int[values.Length];
-            Array.Copy(values, 0, m_array, 0, values.Length);
+            values.AsSpan().CopyTo(m_array);
```

Comment:
Is this measurably faster?  If yes, can we fix Array.Copy so that's not the case?  And if we can't, should we be replacing all such Array.Copy usage with span throughout coreclr/corefx?

Reply:
> Is this measurably faster?

No, not really. It is only faster for medium input lengths (32 to 256). Anything above/below, Array.CopyTo is similar or better.

> If no, let's minimize the unnecessary churn.

OK. Will revert.

```C#
[SimpleJob(warmupCount: 5, targetCount: 10)]
public class ArrayOrSpanCopyTo
{
    [Params(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 16, 32, 64, 99, 128, 256, 512, 999, 1024, 2048, 65536)]
    public int Size { get; set; }

    private int[] _ints;
    private int[] _destination;
    private static Random s_random = new Random(42);

    [GlobalSetup]
    public void Setup()
    {
        _ints = new int[Size];
        _destination = new int[Size];
        for (int i = 0; i < _ints.Length; i++)
        {
            _ints[i] = s_random.Next();
        }
    }

    [Benchmark(Baseline = true)]
    public void ArrayCopyTo()
    {
        Array.Copy(_ints, 0, _destination, 0, _ints.Length);
    }

    [Benchmark]
    public void SpanCopyTo()
    {
        _ints.AsSpan().CopyTo(_destination);
    }
}
```

``` ini

BenchmarkDotNet=v0.11.2, OS=Windows 10.0.17763.107 (1809/October2018Update/Redstone5)
Intel Core i7-6700 CPU 3.40GHz (Skylake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=3.0.100-alpha1-009706
  [Host]     : .NET Core ? (CoreCLR 4.6.26919.02, CoreFX 4.6.26919.02), 64bit RyuJIT
  DefaultJob : .NET Core 2.1.3 (CoreCLR 4.6.26725.06, CoreFX 4.6.26725.05), 64bit RyuJIT


```
|      Method | Size |     Mean |     Error |    StdDev | Ratio | RatioSD |
|------------ |----- |---------:|----------:|----------:|------:|--------:|
| **ArrayCopyTo** |    **1** | **10.90 ns** | **0.1125 ns** | **0.1052 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    1 | 12.08 ns | 0.1480 ns | 0.1312 ns |  1.11 |    0.02 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **2** | **10.90 ns** | **0.0836 ns** | **0.0698 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    2 | 11.95 ns | 0.0993 ns | 0.0880 ns |  1.10 |    0.01 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **3** | **11.20 ns** | **0.1343 ns** | **0.1256 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    3 | 12.13 ns | 0.1960 ns | 0.1834 ns |  1.08 |    0.02 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **4** | **10.92 ns** | **0.1041 ns** | **0.0923 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    4 | 11.85 ns | 0.0606 ns | 0.0506 ns |  1.09 |    0.01 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **5** | **25.20 ns** | **0.3379 ns** | **0.3161 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    5 | 21.69 ns | 0.1555 ns | 0.1378 ns |  0.86 |    0.01 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **6** | **15.34 ns** | **0.3343 ns** | **0.5003 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    6 | 15.13 ns | 0.1199 ns | 0.1122 ns |  0.96 |    0.03 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **7** | **10.44 ns** | **0.1281 ns** | **0.1136 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    7 | 12.01 ns | 0.1052 ns | 0.0933 ns |  1.15 |    0.01 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **8** | **11.24 ns** | **0.2198 ns** | **0.1836 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    8 | 11.95 ns | 0.1183 ns | 0.1107 ns |  1.06 |    0.02 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |    **9** | **11.85 ns** | **0.1245 ns** | **0.1165 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    9 | 12.44 ns | 0.1182 ns | 0.1106 ns |  1.05 |    0.01 |
|             |      |          |           |           |       |         |
| **ArrayCopyTo** |   **10** | **11.60 ns** | **0.2485 ns** | **0.2324 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |   10 | 12.40 ns | 0.1244 ns | 0.1039 ns |  1.07 |    0.02 |
|             |      |          |           |           |       |         |

|      Method |  Size |        Mean |      Error |     StdDev | Ratio | RatioSD |
|------------ |------ |------------:|-----------:|-----------:|------:|--------:|
| **ArrayCopyTo** |    **16** |    **12.11 ns** |  **0.1669 ns** |  **0.0993 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    16 |    13.08 ns |  1.2233 ns |  0.8091 ns |  1.07 |    0.07 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |    **32** |    **14.32 ns** |  **1.0069 ns** |  **0.5992 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    32 |    13.76 ns |  0.3587 ns |  0.2373 ns |  0.96 |    0.04 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |    **64** |    **25.50 ns** |  **0.4696 ns** |  **0.2456 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    64 |    16.33 ns |  1.7889 ns |  1.1832 ns |  0.65 |    0.05 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |    **99** |    **26.02 ns** |  **0.5151 ns** |  **0.3407 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |    99 |    17.22 ns |  0.1131 ns |  0.0591 ns |  0.66 |    0.01 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |   **128** |    **28.72 ns** |  **0.1956 ns** |  **0.1164 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |   128 |    19.08 ns |  0.2054 ns |  0.1222 ns |  0.66 |    0.01 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |   **256** |    **30.59 ns** |  **0.2609 ns** |  **0.1553 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |   256 |    28.49 ns |  0.3315 ns |  0.1734 ns |  0.93 |    0.01 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |   **512** |    **51.45 ns** |  **1.0055 ns** |  **0.6651 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |   512 |    54.79 ns |  1.8849 ns |  1.2467 ns |  1.07 |    0.03 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |   **999** |    **80.12 ns** |  **1.2728 ns** |  **0.7574 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |   999 |    83.86 ns |  0.6412 ns |  0.3816 ns |  1.05 |    0.01 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |  **1024** |    **88.94 ns** |  **7.5275 ns** |  **4.9789 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |  1024 |   100.05 ns |  5.1750 ns |  3.4229 ns |  1.13 |    0.09 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** |  **2048** |   **148.46 ns** |  **7.7266 ns** |  **5.1107 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo |  2048 |   175.57 ns |  9.0965 ns |  6.0168 ns |  1.18 |    0.06 |
|             |       |             |            |            |       |         |
| **ArrayCopyTo** | **65536** | **8,108.93 ns** | **72.9956 ns** | **48.2821 ns** |  **1.00** |    **0.00** |
|  SpanCopyTo | 65536 | 8,207.54 ns | 77.3221 ns | 51.1438 ns |  1.01 |    0.01 |